### PR TITLE
Keep `@(Reference)` metadata when creating `@(ProjectReference)` items

### PR DESCRIPTION
**Pull request by dougbu**
- #20818, fix e.g. references to Microsoft.Web.Xdt.Extensions in our packages
- make `@(Reference)` items much more broadly applicable
- emit an error when `@(ProjectReference)` is used instead of `@(Reference)`
    - then get rid of the errors (!!)
- rename a couple of projects to match their assembly names
    - then regenerate the `@(ProjectReferenceProvider)` items
- switch intersection approach from Exclude / Exclude to Copy / Update / Copy

Projects of particular interest:
- src/DefaultBuilder/src/Microsoft.AspNetCore.csproj
    - honouring metadata left e.g. Microsoft.AspNetCore.Components.WebAssembly.DevServer package unchanged
    - removed redundant metadata after that confirmation
- src/Razor/tools/Microsoft.AspNetCore.Razor.Internal.Transport/
    - content of this package unchanged but metadata avoids extra work
    - add a comment about the extra work
- src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/
    - success! removes Microsoft.Web.Xdt.Extensions dependency from the package
- src/SiteExtensions/Runtime/Microsoft.AspNetCore.Runtime.SiteExtension.pkgproj
    - add a `Condition` to avoid an ordering issue I hit here
- src/Tools/Extensions.ApiDescription.Server/src/
    - avoid errors the new build ordering and timing caused

Separately, up the timeout in the `<DownloadFile />` task
- hit repeated timeouts downloading dotnet-runtime-5.0.0-rc.1.20370.4-win-x64.zip

nits:
- remove dupe `@(Reference)` item in Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj
- remove useless `%(ProjectReference.IsImplicitlyDefined)` metadata as well as its misspellings
- remove extra spaces from ProjectReferences.props
- clean up a few comments in ResolveReferences.targets

### CODE REVIEW

**Comment by dougbu** [comment]
Also building in https://dev.azure.com/dnceng/internal/_build/results?buildId=753395 to create the Microsoft.AspNetCore.AzureAppServices.SiteExtension package.

**Comment by dougbu** [comment]
Another internal build at https://dev.azure.com/dnceng/internal/_build/results?buildId=754565

**Comment by dougbu** [comment]
I have double-checked the packages for updated projects and found no differences other than the expected assembly file versions and the #20818 fix. This PR is ready for review.

**Comment by dougbu** [comment]
questions about current `@(PackageReference)` items:
- why does Internal.AspNetCore.Analyzers project use Microsoft.CodeAnalysis.CSharp.Workspaces v2.8.0?
    - @JunTaoLuo @pranavkm ❔
- should we move from Yarn.MSBuild 1.15.2 to 1.22.4 i.e. `yarn` 1.13.0 to 1.22.4?
    - @javiercn @BrennanConroy ❔
- should `AddContent` target in Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj instead get 5.0 packages?
    - @BrennanConroy ❔
- should we move `MsQuicPackage.*` version from QuicSampleClient and QuicSampleApp projects to eng/Versions.props?
    - @jkotalik ❔

**Comment by jkotalik** [comment]
> MsQuicPackage.*

Sure.

**Comment by pranavkm** [comment]
> why does Internal.AspNetCore.Analyzers project use Microsoft.CodeAnalysis.CSharp.Workspaces v2.8.0

I'm not really familiar with that package. @JunTaoLuo would know best

**Comment by BrennanConroy** [comment]
>     * should `AddContent` target in Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj instead get 5.0 packages?

I'm not really sure what that part is for, but it does look like we should probably update the version.

>     * should we move from Yarn.MSBuild 1.15.2 to 1.22.4 i.e. `yarn` 1.13.0 to 1.22.4?

No objections here.

File: eng/targets/ResolveReferences.targets [hunk 1]
```diff
@@ -91,24 +88,55 @@
...
-    <ProjectReference Include="@(_ProjectReferenceByAssemblyName->'%(ProjectPath)')" />
-    <Reference Remove="@(_ProjectReferenceByAssemblyName)" />
+  <ItemGroup Condition=" '$(EnableCustomReferenceResolution)' == 'true' AND '$(UseProjectReferences)' == 'true' ">
+    <!-- Copy then Update / Copy to intersect the ProjectReferenceProvider and Reference item groups. -->
+    <_AllProjectReference Include="@(ProjectReferenceProvider)" />
+
+    <!-- Use only Reference items when project is a reference provider. Simplifies project moves and shortens files. -->
+    <_AllProjectReference Update="@(ProjectReference->'%(Filename)')" DirectUse="1" />
+
+    <_AllProjectReference Update="@(Reference)" Use="1">
```
Comment (wtgodbe) [inline eng/targets/ResolveReferences.targets:104]: Is there a guarantee we'll never need to update this list of metadata?

**Review by wtgodbe — APPROVED** [review]
Seems like goodness

File: eng/targets/ResolveReferences.targets (hunk 1 above)
Comment (dougbu) [reply eng/targets/ResolveReferences.targets:104]: There are few guarantees in life😈 But, the current list goes pretty far beyond what we're likely to use in `*.csproj` and `*.fsproj` projects and should hold us in good stead until and unless new metadata is invented. I included everything I saw used in existing `@(Reference)` and `@(ProjectReference)` items in this repo as well as what seemed plausible from the documentation.

**Comment by dougbu** [comment]
I'll file an issue about the changes we could or should make based on the answers to my [questions above](https://github.com/dotnet/aspnetcore/pull/21567#issuecomment-668267663) once we hear from @JunTaoLuo.

**Comment by JunTaoLuo** [comment]
FYI, after our offline discussion, we're going to try updating the TFM of Internal.AspNetCore.Analyzers to target netstandard2.0 TFM so it can consume the same version of the roslyn packages as the rest of the repo. Also good to double check if we plan on removing this package outright at some point.

**Comment by dougbu** [comment]
@pranavkm following up on @JunTaoLuo's comment just above
> Also good to double check if we plan on removing this package outright at some point.

I'm double-checking😺 I couldn't find a seemingly-relevant issue about removing the analyzer but also can't find anywhere we're suppressing it other than for analyzers and tools. Do we need this anymore❔

And, if we need it, why is it packaged i.e. is the package just a holdover from when the project was in dotnet/extensions❔

**Comment by pranavkm** [comment]
> And, if we need it, why is it packaged i.e. is the package just a holdover from when the project was in dotnet/extensions❔

That seems like the most likely scenario. 

>  I couldn't find a seemingly-relevant issue about removing the analyzer 
I think I commented on this in another issue. There's only one analyzer in the package about not exposing pubternal types. We don't have any of those any more so it seems entirely unnecessary.

**Comment by dougbu** [comment]
#24608 covers the follow-up points above

**Pull request merged**